### DISCUSSION TITLE
How to send a post with hashtags?

### DISCUSSION THREAD

**Discussion by Murray2015** [MarshalX/atproto#507] [Questions] [answered]
Hi, I'm trying to create a post with a hashtag, which I _think_  I don't need to do anything special to get to work. But they don't work - the hashtags don't post as hashtags; they are plain text. Example below: 

```python 
    bsky_client = Client()
    bsky_client.login(bsky_username, bsky_password)
    bsky_client.send_post("I love #cats", langs=["en-UK"])
```

Posts `"I love #cats"`, but as plain text not a hashtag. I don't know if this is expected behaviour, in which case, could someone please point me to an example of how to make this work? Otherwise, it may be a bug 🙂

**Comment by MarshalX** [comment, marked as answer]
Hi, you should work with facets (rich text). Here you can read more about it: https://docs.bsky.app/docs/advanced-guides/post-richtext

Here is how to use SDK`s helper to make it easy:
```python
from atproto import client_utils
text = client_utils.TextBuilder().text('I love ').tag('#cats', 'cats')
client.send_post(text)
```

Read more about TextBuilder: https://atproto.blue/en/latest/atproto_client/utils/text_builder.html

**Comment by Murray2015** [comment]
Thanks @MarshalX, that works a treat!
